packages/core/src/errors.ts

```diff
@@ -1,58 +1,176 @@
  *
- * * @alpha
+ * @alpha
  */
-export class IgnitionValidationError extends IgnitionError {
-  constructor(message: string) {
+export class IgnitionPluginError extends CustomError {
+  public static isIgnitionPluginError(
+    error: any
+  ): error is IgnitionPluginError {
+    return (
+      typeof error === "object" &&
+      error !== null &&
+      (error as IgnitionPluginError)._isIgnitionPluginError === true
+    );
+  }
+
+  private readonly _isIgnitionPluginError = true;
+
+  public readonly pluginName: string;
+
+  constructor(pluginName: string, message: string) {
     super(message);
+    this.pluginName = pluginName;
+    Object.setPrototypeOf(this, IgnitionPluginError.prototype);
+  }
+}
 
-    // This is required to allow calls to `resetStackFrom`,
-    // otherwise the function is not available on the
-    // error instance
-    Object.setPrototypeOf(this, new.target.prototype);
-  }
-
-  /**
-   * Reset the stack hiding parts that are bellow the given function.
-   * The intention is the function should be part of the user callable
-   * api, so that the stack leads directly to the line in the module
-   * the user called (i.e. `m.contract(...)`)
-   *
-   * @param f - the function to hide all of the stacktrace above
-   *
-   * @internal
-   */
-  public resetStackFrom(f: () => any) {
-    Error.captureStackTrace(this, f);
+/**
+ * This class is used to throw errors from *core* Ignition plugins.
+ * If you are developing a third-party plugin, use IgnitionPluginError instead.
+ *
+ * @alpha
+ */
+export class NomicIgnitionPluginError extends IgnitionPluginError {
+  public static isNomicIgnitionPluginError(
+    error: any
+  ): error is NomicIgnitionPluginError {
+    return (
+      typeof error === "object" &&
+      error !== null &&
+      (error as NomicIgnitionPluginError)._isNomicIgnitionPluginError === true
+    );
+  }
+
+  private readonly _isNomicIgnitionPluginError = true;
+
+  constructor(pluginName: string, message: string) {
+    super(pluginName, message);
+    Object.setPrototypeOf(this, NomicIgnitionPluginError.prototype);
   }
 }
 
 /**
- * This error class is thrown in situations where Ignition
- * intentionally doesn't support a certain operation.
+ * This function applies error messages templates like this:
+ *
+ *  - Template is a string which contains a variable tags. A variable tag is a
+ *    a variable name surrounded by %. Eg: %plugin1%
+ *  - A variable name is a string of alphanumeric ascii characters.
+ *  - Every variable tag is replaced by its value.
+ *  - %% is replaced by %.
+ *  - Values can't contain variable tags.
+ *  - If a variable is not present in the template, but present in the values
+ *    object, an error is thrown.
+ *
+ * @param template - The template string.
+ * @param values - A map of variable names to their values.
  *
  * @alpha
  */
-export class UnsupportedOperationError extends IgnitionError {
```

Comment:
This error was only being used in one place, so I removed it as well and added the message for it to `errors-list.ts`